mithril-aggregator/src/database/repository/immutable_file_digest_repository.rs

```diff
@@ -0,0 +1,434 @@
+            .ok_or_else(|| panic!("Upserting an immutable_file_digest should not return nothing."))
+    }
+
+    /// Delete all [ImmutableFileDigestRecord] from the database.
+    pub async fn delete_all(&self) -> StdResult<()> {
+        self.connection
+            .fetch_first(DeleteImmutableFileDigestQuery::all())?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl ImmutableFileDigestCacheProvider for ImmutableFileDigestRepository {
+    async fn store(
+        &self,
+        digest_per_filenames: Vec<(ImmutableFileName, HexEncodedDigest)>,
+    ) -> CacheProviderResult<()> {
+        for (filename, digest) in digest_per_filenames {
+            self.upsert_immutable_file_digest(&filename, &digest)
+                .await
+                .map_err(ImmutableDigesterCacheStoreError::StoreError)?;
+        }
+
+        Ok(())
+    }
+
+    async fn get(
+        &self,
+        immutables: Vec<ImmutableFile>,
+    ) -> CacheProviderResult<BTreeMap<ImmutableFile, Option<HexEncodedDigest>>> {
+        let mut result = BTreeMap::new();
+        for immutable in immutables {
+            let immutable_file_digest = self
+                .get_immutable_file_digest(&immutable.filename)
+                .await
+                .map_err(ImmutableDigesterCacheGetError::StoreError)?;
+
+            result.insert(immutable, immutable_file_digest.map(|f| f.digest));
+        }
+
+        Ok(result)
+    }
+
+    async fn reset(&self) -> CacheProviderResult<()> {
+        self.delete_all()
+            .await
+            .map_err(ImmutableDigesterCacheGetError::StoreError)?;
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl ImmutableFileDigestMapper for ImmutableFileDigestRepository {
+    async fn get_immutable_file_digest_map(
+        &self,
+    ) -> StdResult<BTreeMap<ImmutableFileName, HexEncodedDigest>> {
+        let immutable_file_digest_map = BTreeMap::from_iter(
+            self.get_all_immutable_file_digest()
+                .await?
+                .into_iter()
+                .map(|record| (record.immutable_file_name, record.digest)),
+        );
+
+        Ok(immutable_file_digest_map)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::database::test_helper::main_db_connection;
+
+    use super::*;
+
+    async fn get_connection() -> Arc<SqliteConnection> {
+        let connection = main_db_connection().unwrap();
+
+        Arc::new(connection)
+    }
+
+    mod repository {
+        use super::*;
+
+        #[tokio::test]
+        async fn repository_get_immutable_file_digest() {
+            let repository = ImmutableFileDigestRepository::new(get_connection().await);
+            let immutable_file_name: ImmutableFileName = "123.chunk".to_string();
+            let digest = "digest-123";
+
+            let immutable_file_digest_result = repository
+                .get_immutable_file_digest(&immutable_file_name)
+                .await
+                .unwrap();
+            assert!(immutable_file_digest_result.is_none());
```

Comment:
I would prefer using `assert_eq` here as it have the benefit to output what was found when the assertion fail.
```suggestion
            assert_eq!(None, immutable_file_digest_result);
```

Reply:
Done